File: lesson_20/src/main/java/lesson20/service/ProductService.java
package lesson20.service;

import lesson20.entities.Product;
import lesson20.repository.ProductRepository;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class ProductService {
    ProductRepository productRepository;
    ProductService(ProductRepository productRepository) {
        this.productRepository = productRepository;
    }
    public void create(Product product) {
        productRepository.create(product);
    }

    public Product getById(Long id) {
        return productRepository.getById(id);
    }

    public List<Product> getAll() {
        return productRepository.getAll();
    }
}

File: lesson_20/src/main/kotlin/lesson20kt/services/ProductService.kt
package lesson20kt.services

import lesson20kt.entities.Product
import lesson20kt.repositories.ProductRepository
import org.springframework.stereotype.Component

@Component
class ProductService(val prodectRepository: ProductRepository) {

    fun save(product: Product) = prodectRepository.save(product)
    fun load(id: Long): Product = prodectRepository.load(id)
    fun loadAll(): List<Product> = prodectRepository.loadAll()
}